File: src/main/java/org/example/TestExtraEntryPoints.java
package org.example;

import pascal.taie.World;
import pascal.taie.analysis.pta.core.solver.DeclaredParamProvider;
import pascal.taie.analysis.pta.core.solver.EntryPoint;
import pascal.taie.analysis.pta.core.solver.Solver;
import pascal.taie.analysis.pta.plugin.Plugin;
import pascal.taie.language.classes.JClass;
import pascal.taie.language.classes.JMethod;

public class TestExtraEntryPoints implements Plugin {
    private Solver theSolver;

    @Override
    public void setSolver(Solver solver) {
        if (solver != null) {
            theSolver = solver;
        }
        Plugin.super.setSolver(solver);
    }

    @Override
    public void onStart() {
        for (JClass cls : World.get().getClassHierarchy().allClasses().toList()) {
            if (!cls.getName().equals("Main")) continue;
            for (JMethod method : cls.getDeclaredMethods()) {
                if (method.getName().startsWith("entry")) {
                    theSolver.addEntryPoint(new EntryPoint(
                            method,
                            new DeclaredParamProvider(method, theSolver.getHeapModel(), 2)
                    ));
                }
            }
        }
    }
}

File: src/main/kotlin/org/example/TestExtraEntryPoints.kt
package org.example

import pascal.taie.World
import pascal.taie.analysis.pta.core.solver.DeclaredParamProvider
import pascal.taie.analysis.pta.core.solver.EntryPoint
import pascal.taie.analysis.pta.core.solver.Solver
import pascal.taie.analysis.pta.plugin.Plugin

class TestExtraEntryPoints : Plugin {
    private lateinit var theSolver: Solver

    override fun setSolver(solver: Solver?) {
        if (solver != null) {
            theSolver = solver
        }
        super.setSolver(solver)
    }

    override fun onStart() {
        for (cls in World.get().classHierarchy.allClasses()) {
            if (cls.name != "Main") continue
            for (method in cls.declaredMethods) {
                if (method.name.startsWith("entry")) {
                // if (method.name == "entry") {
                    theSolver.addEntryPoint(EntryPoint(
                        method,
                        DeclaredParamProvider(method, theSolver.heapModel, 2)
                    ))
                }
            }
        }
    }
}
